### PR TITLE
Resolve compiler warnings for DataSeries on boost 1.65

### PR DESCRIPTION
**Pull request by lvkv**
Same as before, I've attached `make` output containing the warnings we're trying to resolve below.
[dataseries_make_output.txt](https://github.com/sbu-fsl/DataSeries/files/5343943/dataseries_make_output.txt)

### CODE REVIEW

**Comment by lvkv** [comment]
A summary of changes and their reasoning.

**Tl;dr** no code changes were necessary so we're suppressing all these warnings:

- The "set but unused" warning is a false alarm - the reason the variable is "unused" is because we're trying to catch an error that might be thrown by creating it. I suppressed that warning.
- There were 2 `locally defined but not used` warnings. One of them was a false alarm, so I suppressed this warning as well.
- The `compiler can assume that the address of ‘series’ will never be NULL` warnings seem very fishy to me. It didn't feel right to replace all occurrences of `SINVARIANT(&series != NULL)` with `SINVARIANT(true)`, so I suppressed the warning.

Also, I have an extension that automatically removes trailing whitespace on save. I didn't realize I had it enabled, so some of these commits also remove trailing whitespace in files. I don't think this should matter.
